src/languageservice/services/yamlOnTypeFormatting.ts

```diff
@@ -45,5 +45,14 @@ export function doDocumentOnTypeFormatting(
       return [TextEdit.insert(position, '  ')];
     }
   }
+
+  if (params.ch === '\t' && params.options.insertSpaces) {
```

Comment:
I think the condition should be refined so that the replacement only happens before the line payload (ie no previous character on line or all previous chars on line are whitespaces)